specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,190 @@
+
+### Plan for Nimble-Provided Labels
+
+Conceptually the localized strings are very similar to the sharing pattern of Design Tokens. The vast majority of the time, you want to use the same value (i.e. numeric increment/decrement button text or collapse button text is not control instance specific) but want to adjust to a global config, i.e. theme or language. But also want to be able to override for specific controls as needed.
+
+We will create non-CSS-property design tokens, similar to the `theme` and `direction` design tokens. The values can be set via the `nimble-theme-provider`:
+
+```html
+<nimble-theme-provider
+    label-number-field-decrement="Increment"
+    label-number-field-increment="Decrement"
+    label-table-collapse-all="Collapse All"
+>
+<nimble-table></nimble-table>
+</nimble-theme-provider>
+```
+
+Pros:
+
+- Easy discovery of localized string configuration. They are all on the theme-provider.
+- Can configure once for the whole page in one spot, does not need to be per control. Much fewer duplicated attributes / slotted elements in the page.
+- Can override for specific elements as needed by wrapping in a theme-provider.
+- Can configure to be attribute or slotted on the theme provider. But if don't need mixed content then attribute makes sense.
+- User provided control-specific content should still be provided via attributes / slots. This is for control provided content (the table control defines the collapse button, not the user).
+
+Cons: 
+
+- Still somewhat verbose in the page, to have all of the localized strings as attributes
+    - We can consider using the `fromView` attribute mode on the attributes for the strings on the theme-provider. If clients set the strings using the theme-provider properties rather than as attributes in the HTML (as is currently proposed for Angular), `fromView` would not reflect those strings back to the DOM.
+- Not an originally intended use case for DesignToken
+- Only nimble-components defined within Nimble can add new attributes to the theme-provider.
+  - We're encouraging clients to add new components to Nimble, including custom column types for the table, and those can use this approach
+  - Components outside Nimble could still have accessible labels by defining attributes on their components for them
+
+#### Implementation Details
+
+See the prototype branch: [localizable-labels-prototype](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b)
+
+**nimble-components**  
+We'll create a file containing the label IDs and English string values. A simple version of this is a JSON file (Prototype: [`labels.json`](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-0e77bd18c80fb68ad9624a1eeccbec4a539f98b92e2ff97a013f763cc897a8ce)):
```

Comment:
I somewhat agree with not wanting to lean too heavily on code generation. The current prototype definitely results in a fair amount of TypeScript code hidden/ obscured in the generator source files which makes those more complicated/ hard to maintain.

On the other hand I also don't want devs to have to hand-edit multiple different files and copy-and-paste the attribute/changed() functions over and over again, since that can lead to copy/paste errors too. I'll have to see if there's a better middle ground.

Reply:
Prototype with the new proposal with the `nimble-i18n-*` providers uses no codegen.

I think we'd probably still want to codegen the Angular and Blazor wrappers, so said that in the HLD.